common/djangoapps/course_category/admin.py

```diff
@@ -7,6 +7,32 @@
 from models import CourseCategory, Program
 
 
+class ProductModelMultipleChoiceField(forms.ModelMultipleChoiceField):
+    '''
+    Redefinition of clean method.
+
+    Values that original method recieves by m2m from CourseOverview are course_id (str),
+    while CourseLocator needed.
+    '''
+    def clean(self, value):
+        value = map(CourseKey.from_string, value)
+        return super(ProductModelMultipleChoiceField, self).clean(value)
+
+
+class ProgramForm(forms.ModelForm):
+    products = ProductModelMultipleChoiceField(queryset=CourseOverview.objects.filter(product=True), required=False)
+
+    def __init__(self, *args, **kwargs):
```

Comment:
Why do we need to override a constructor here?

Reply:
@cmltaWt0 I guess we can remove this empty constructor from here? what do you think?

Reply:
@drtwisted Sure. I totally agree.